src/main/java/core/basesyntax/strategy/OperationStrategyImpl.java

```diff
@@ -0,0 +1,27 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.Operation;
+import core.basesyntax.strategy.operations.BalanceOperationHandler;
+import core.basesyntax.strategy.operations.OperationHandler;
+import core.basesyntax.strategy.operations.PurchaseOperationHandler;
+import core.basesyntax.strategy.operations.ReturnOperationHandler;
+import core.basesyntax.strategy.operations.SupplyOperationHandler;
+import java.util.HashMap;
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private final Map<Operation, OperationHandler> operationHandlerMap;
+
+    public OperationStrategyImpl() {
+        operationHandlerMap = new HashMap<>();
+        operationHandlerMap.put(Operation.BALANCE, new BalanceOperationHandler());
+        operationHandlerMap.put(Operation.PURCHASE, new PurchaseOperationHandler());
+        operationHandlerMap.put(Operation.RETURN, new ReturnOperationHandler());
+        operationHandlerMap.put(Operation.SUPPLY, new SupplyOperationHandler());
```

Comment:
```suggestion
    public OperationStrategyImpl(Map<Operation, OperationHandler> operationHandlerMap) {
       this. operationHandlerMap = operationHandlerMap;
```
You are already creating this map in the Main class. Just pass it here via the constructor